apps/api/src/applicant/ienapplicant.util.service.ts

```diff
@@ -176,12 +177,15 @@ export class IENApplicantUtilService {
     if (!statusObj) {
       throw new NotFoundException(`Status with given value "${status}" not found`);
     }
-    if (statusObj) {
-      if (statusObj.parent?.id != 10003) {
-        throw new BadRequestException(
-          `Only recruitment-related milestones/statuses are allowed here`,
-        );
-      }
+
+    if (user.roles.some(r => r.slug === RoleSlug.Admin)) {
+      return statusObj;
+    }
+
+    if (statusObj && !user.ha_pcn_id && statusObj.parent?.id != 10003) {
```

Comment:
We'd better move this logic to caller because `getStatusById` should be the atomic method returning just status by id.